packages/app/obojobo-document-engine/src/scripts/oboeditor/components/editor-app.js

```diff
@@ -123,6 +126,30 @@ class EditorApp extends React.Component {
 		this.reloadDraft(this.state.draftId, mode)
 	}
 
+	displayLockedModal() {
+		this.setState({
+			requestStatus: 'invalid',
+			requestError: {
+				title: 'Module is Being Edited.',
+				message: 'Someone else is currently editing this module, please try again later.'
+			}
+		})
+	}
+
+	startRenewEditLockInterval() {
+		setInterval(() => {
+			this.createEditLock(this.state.draftId).catch(() => {
+				this.displayLockedModal()
```

Comment:
This is a catch all for every error. If someone no longer has permissions to a module then they are shown this error, when in reality the error they are getting is `"Authoring permissions required to create a lock"`.

Reply:
That's true.  Since this is now the first thing to check your access rights in the editor it'll need to tell you if you just don't have permission.  I might be able to use the status to determine this